src/components/Calculator.js

```diff
@@ -1,36 +1,55 @@
-const Calculator = () => (
-  <div className="calculator">
-    <div className="display">0</div>
-    <div className="row">
-      <button type="button" className="btn clear">AC</button>
-      <button type="button" className="btn plus-minus">+/-</button>
-      <button type="button" className="btn percent">%</button>
-      <button type="button" className="btn operator">&divide;</button>
-    </div>
-    <div className="row">
-      <button type="button" className="btn number">7</button>
-      <button type="button" className="btn number">8</button>
-      <button type="button" className="btn number">9</button>
-      <button type="button" className="btn operator">&times;</button>
-    </div>
-    <div className="row">
-      <button type="button" className="btn number">4</button>
-      <button type="button" className="btn number">5</button>
-      <button type="button" className="btn number">6</button>
-      <button type="button" className="btn operator">&minus;</button>
-    </div>
-    <div className="row">
-      <button type="button" className="btn number">1</button>
-      <button type="button" className="btn number">2</button>
-      <button type="button" className="btn number">3</button>
-      <button type="button" className="btn operator">+</button>
-    </div>
-    <div className="row">
-      <button type="button" className="btn zero number">0</button>
-      <button type="button" className="btn decimal">.</button>
-      <button type="button" className="btn equals operator">=</button>
+import React, { useState } from 'react';
+import Button from './Button';
+import calculate from './logic/calculate';
+
+const Calculator = () => {
+  const [calculatorData, setCalculatorData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (buttonName) => {
+    const newData = calculate(calculatorData, buttonName);
+    setCalculatorData(newData);
+  };
+
+  const displayValue = calculatorData.next || calculatorData.total || '0';
+
+  return (
+    <div className="calculator">
+      <div className="display">{displayValue}</div>
+      <div className="row">
+        <Button className="clear" text="AC" onClick={() => handleClick('AC')} />
+        <Button className="plus-minus" text="+/-" onClick={() => handleClick('+/-')} />
+        <Button className="percent" text="%" onClick={() => handleClick('%')} />
+        <Button className="operator" text="÷" onClick={() => handleClick('÷')} />
+      </div>
+      <div className="row">
+        <Button className="number" text="7" onClick={() => handleClick('7')} />
+        <Button className="number" text="8" onClick={() => handleClick('8')} />
+        <Button className="number" text="9" onClick={() => handleClick('9')} />
+        <Button className="operator" text="×" onClick={() => handleClick('×')} />
```

Comment:
- [ ] In operate.js, the expected sign for multiplications is `x` but in your Calculator.js you are using `×` which leads to an error. kindly fix it
<img width="763" alt="image" src="https://user-images.githubusercontent.com/37901469/230716271-f509bbda-7197-46a4-b9a0-8cbdeb80fc09.png">

Reply:
Hi @m05e5, 
Thanks for reviewing my project and mentioning the error.
I have just modified the multiplication sign in [`Calculator.js`](https://github.com/ab-noori/math-magicians/blob/6399e96b5128b7d02dbe8b659f25e899d71de1cd/src/components/Calculator.js#L32) component.